src/astarte-client/client.ts

```diff
@@ -17,101 +17,138 @@
 */
 
 import axios from 'axios';
-import jwt from 'jsonwebtoken';
+import Phoenix from 'phoenix';
 
-import Block from './models/Block';
+import { AstarteBlock, AstarteCustomBlock, toAstarteBlock } from './models/Block';
 
-const { Socket } = require('phoenix');
+import { AstarteBlockDTO, AstarteJWT } from './types';
+
+type Channel = any;
+type Trigger = any;
 
 // Wrap phoenix lib calls in promise for async handling
-function openNewSocketConnection(connectionParams, onErrorHanlder, onCloseHandler) {
+async function openNewSocketConnection(
+  connectionParams: any,
+  onErrorHanlder: any,
+  onCloseHandler: any,
+): Promise<Phoenix.Socket> {
   const { socketUrl, realm, token } = connectionParams;
 
   return new Promise((resolve) => {
-    const phoenixSocket = new Socket(socketUrl, {
+    const phoenixSocket = new Phoenix.Socket(socketUrl, {
       params: {
         realm,
         token,
       },
     });
-    phoenixSocket.onError((e) => onErrorHanlder(e));
-    phoenixSocket.onClose((e) => onCloseHandler(e));
+    phoenixSocket.onError((e: any) => onErrorHanlder(e));
+    phoenixSocket.onClose((e: any) => onCloseHandler(e));
     phoenixSocket.onOpen(() => {
       resolve(phoenixSocket);
     });
     phoenixSocket.connect();
   });
 }
 
-function joinChannel(phoenixSocket, channelString) {
+async function joinChannel(phoenixSocket: any, channelString: string): Promise<Channel> {
   return new Promise((resolve, reject) => {
     const channel = phoenixSocket.channel(channelString, {});
     channel
       .join()
       .receive('ok', () => {
         resolve(channel);
       })
-      .receive('error', (err) => {
+      .receive('error', (err: any) => {
         reject(err);
       });
   });
 }
 
-function leaveChannel(channel) {
+async function leaveChannel(channel: Channel): Promise<void> {
```

Comment:
If i recall correctly, using async functions automatically returns a promise, even if you don't specify it.
I'd leave that out of the PR to avoid introducing unexpected bugs.

Reply:
`leaveChannel` already returns a Promise so it's no harm to annotate the function as `async`: i.e. you won't get a Promise of a Promise since promises auto-unwrap, `async` it's just annotation to keep the same convention of all the other functions returning promises.